lib/saluki-components/src/sources/heartbeat/mod.rs

```diff
@@ -0,0 +1,94 @@
+use std::time::Duration;
+
+use async_trait::async_trait;
+use memory_accounting::{MemoryBounds, MemoryBoundsBuilder};
+use saluki_context::Context;
+use saluki_core::components::{sources::*, ComponentContext};
+use saluki_core::topology::OutputDefinition;
+use saluki_error::GenericError;
+use saluki_event::{metric::Metric, DataType, Event};
+use tokio::{select, time::interval};
+use tracing::{debug, error};
+
+/// Heartbeat source.
+///
+/// Emits a "heartbeat" metric on a configurable interval.
+#[derive(Clone, Debug)]
+pub struct HeartbeatConfiguration {
+    /// Interval for heartbeat metrics in seconds
+    pub heartbeat_interval_secs: u64,
+}
+
+impl Default for HeartbeatConfiguration {
+    fn default() -> Self {
+        Self {
+            heartbeat_interval_secs: 10,
+        }
+    }
+}
+
+/// Source that doesn't produce any events or optionally emits heartbeat metrics.
+struct HeartbeatSource {
```

Comment:
```suggestion
struct Heartbeat {
```

(Just making this match how other components define their primary type.)